### PR TITLE
Features/topology inventory

### PR DESCRIPTION
**Pull request by skochetkov**
- safely replaced using ModelBuilderImpl with InventoryModelManager.
- all tests that using model builder are updated
- provided adapter for transforming different inputs to internal representation of IMM

### CODE REVIEW

File: engine/topology/src/main/java/org/opennms/oce/engine/topology/shell/GenerateGraphML.java [hunk 1]
```diff
@@ -61,7 +62,10 @@
 
     @Override
     public Object execute() throws IOException {
-        final Model model = ModelBuilderImpl.buildModel(inventoryDatasource.getInventory());
```
Comment (j-white) [inline engine/topology/src/main/java/org/opennms/oce/engine/topology/shell/GenerateGraphML.java]: We replace this one line with 4 lines in multiple places. Can we make a similar utility function that accepts the inventory and returns the model instead?

File: engine/topology/src/test/java/org/opennms/oce/engine/topology/model/ModelBuilderImplTest.java [hunk 2]
```diff
@@ -1,103 +0,0 @@
...
-
-import java.util.Set;
-
-import org.junit.Test;
-import org.opennms.oce.driver.test.MockInventory;
-
-public class ModelBuilderImplTest {
-
-    @Test
-    public void canGenerateModel() {
```
Comment (j-white) [inline engine/topology/src/test/java/org/opennms/oce/engine/topology/model/ModelBuilderImplTest.java]: This test can still be used against the model that was built from the sample network. `UpdateInventoryTest` has some coverage, but doesn't include calls to functions like `getPeers()`.

Comment (skochetkov) [reply engine/topology/src/test/java/org/opennms/oce/engine/topology/model/ModelBuilderImplTest.java]: that's actually in TODO items from UpdateInventoryTest. I will write content for this test there.

**Comment by j-white** [comment]
Can we rebase this branch over `features/topology`? There are a lot of unrelated commits here.

**Pull request merged**